lib/LANraragi/Controller/Api/Archive.pm

```diff
@@ -188,11 +218,14 @@ sub create_archive {
 
     my $tempfile = $tempdir . '/' . $filename;
     if ( !$upload->move_to($tempfile) ) {
+        $logger->error("Could not move uploaded file $filename to $tempfile");
+        $redis->del("upload:$filename");
+        $redis->quit();
         return $self->render(
             json => {
                 operation   => "upload",
                 success     => 0,
-                error       => "Couldn't move uploaded file to $tempfile."
+                error       => "Server Error"
```

Comment:
Not the biggest fan of generic "server error" messages which are always unhelpful - especially since we have the reason for the error here.

Reply:
I can change this back. My opinion is that server-side errors should not be exposed to the client for security reasons/necessity, but this obviously is a tradeoff with debugging.

Reply:
Right, I do agree it's better to avoid exposing information about the server file structure or similar.  
Would `Couldn't move uploaded file to temporary location` work as a compromise? The detailed error is already logged server-side so I'd be fine with that.

Reply:
No complaints here, should be reflected in the new commits.